lib/Parser/Parser.cpp

```diff
@@ -725,18 +725,8 @@ bool Parser::typeCheckInst(Inst::Kind IK, unsigned &Width,
 
   switch (IK) {
   case Inst::BSwap:
-    if (Width != 16 && Width != 32 && Width != 64) {
-      ErrStr = "bswap doesn't support " + std::to_string(Width) + " bits";
-      return false;
-    }
-    break;
-  case Inst::CtPop:
-  case Inst::Ctlz:
-  case Inst::Cttz:
-    if (Width !=8 && Width != 16 && Width != 32 &&
-        Width != 64 && Width != 256) {
-      ErrStr = std::string(Inst::getKindName(IK)) + " doesn't support " +
-        std::to_string(Width) + " bits";
+    if (Width % 16 != 0) {
+      ErrStr = "bswap doesn't support odd number of bytes";
```

Comment:
a bit more informative would be "bswap argument must be a multiple of 16 bits"

also make sure it's >0 unless that's done elsewhere

Reply:
yup. > 0 check is already there just above this switch block